### DISCUSSION TITLE
Easiest way to create a custom page with custom HTML code

### DISCUSSION THREAD

**Discussion by mhdaouas** [directus/directus#8568] [Archive]
I want to create a page that is related to many content types (not only one) or no one at all. What is the easiest way to create a custom HTML page with a link in the left sidebar?

I tried embedding an iframe in a single content type detail page with the WYSIWYG element but I couldn't render it without the "standard decorations" (I want a plain iframe with no borders or additional buttons).

**Comment by rijkvanzanten** [comment]
Making a custom module is easiest way to inject a new page with whatever content you'd like:
https://docs.directus.io/guides/modules/

**Reply by mhdaouas** [reply]
I couldn't link my custom module to a menu entry (link) under the project properties page. How should I proceed to see my fresh module in the app? It's not explained in the official documentation.

**Reply by Attacler** [reply]
@mhdaouas assuming you have the extension builded and putted it inside exentions/modules, you can go to project settings and control it there "Module navigation"

**Reply by mhdaouas** [reply]
I saw in the sample code that there is a path with an empty string. I tried replacing it with "customModule" but I still can't access the module by URL. I tried with the URL "host:port/customModule" and "host:port/admin/customModule" without success.

**Reply by mhdaouas** [reply]
Actually my custom module is not loaded at all when I run "yarn directus start". I followed the steps in the official guide and just replaced the path in routes.

**Reply by rijkvanzanten** [reply]
@nickrum mind taking a peak at those guides to see if anything is wrong/missing?

**Reply by mhdaouas** [reply]
Ok I did a mistake: I didn't create a folder with the name of the custom module under extensions/modules folder. Now I'm able to find the module under Project Settings > Modules. When I add the menu entry for the module and click on it, I get a blank screen. I tried with the two module.vue examples in the official custom module creation guide.

**Reply by joselcvarela** [reply]
Hello @mhdaouas 
I've tried locally and seems to be OK.
Though, my first generated file was empty, so check if your `extensions/modules/example/index.js` has content and should be similar to this:
```
import{resolveComponent as o,openBlock as t,createBlock as l,withCtx as e,createVNode as n,createElementBlock as i,Fragment as c,renderList as s,createTextVNode as a,toDisplayString as u}from"vue";var d={data:()=>({collections:null}),methods:{logToConsole:function(){console.log(this.collections)}},inject:["api"],mounted(){console.log(this.api),this.api.get("/collections?limit=-1").then((o=>{this.collections=o.data.data}))}};const m=a("Log collections to console");d.render=function(d,r,f,p,v,h){const g=o("v-list-item"),C=o("v-list"),_=o("v-button"),k=o("private-view");return t(),l(k,{title:"Example Collection List"},{default:e((()=>[n(C,null,{default:e((()=>[(t(!0),i(c,null,s(v.collections,(o=>(t(),l(g,{key:o.collection},{default:e((()=>[a(u(o.collection),1)])),_:2},1024)))),128))])),_:1}),n(_,{onClick:h.logToConsole},{default:e((()=>[m])),_:1},8,["onClick"])])),_:1})},d.__file="module.vue";var r={id:"custom",name:"Custom",icon:"box",routes:[{path:"",component:d}]};export{r as default};
```
By using the example provided here: https://docs.directus.io/guides/modules/#accessing-the-api-from-within-your-extension

**Reply by mhdaouas** [reply]
I used the api example as it is (just added a name of a route instead of empty string in index.js) without success.

**Reply by nickrum** [reply]
@rijkvanzanten the guides seem to be fine.

@mhdaouas That might actually be the cause of your issue. The `path` of your module's main route has to be an empty string in order for the component to be shown when navigating to `/admin/<module-id>`.

**Reply by rslonik** [reply]
I've tested in a rc92 install and module didn't show up in settings. Then I've tried in a rc95 and the module appeared. Last I've update to rc99 and it's not showing again.

Does it work for you in rc99?

**Reply by rslonik** [reply]
It's some cache on the browser. In anonymous tab my module appears.

**Reply by nickrum** [reply]
Interesting! I'm wondering why no one else has experienced this yet. Regardless, we should probably instruct browsers to not cache the extension entrypoints.
